master/buildbot/reporters/stash.py

```diff
@@ -82,3 +90,52 @@ def send(self, build):
                 content = yield response.content()
                 log.error("{code}: Unable to send Stash status: {content}",
                           code=response.code, content=content)
+
+
+class StashPRCommentPush(http.HttpStatusPushBase):
+    name = "StashPRCommentPush"
+
+    @defer.inlineCallbacks
+    def reconfigService(self, base_url, user, password, text=None, verbose=False, **kwargs):
+        yield http.HttpStatusPushBase.reconfigService(self, wantProperties=True,
+                                                      **kwargs)
+        self.text = text or Interpolate('Builder: %(prop:buildername)s Status: %(prop:statustext)s')
+        self.verbose = verbose
+        self._http = yield httpclientservice.HTTPClientService.getService(
+            self.master, base_url, auth=(user, password))
+
+    @defer.inlineCallbacks
+    def send(self, build):
+        if build['complete'] \
+           and build['properties'].has_key("pullrequesturl"):
+                yield self.sendPullRequestComment(build)
+
+    @defer.inlineCallbacks
+    def sendPullRequestComment(self, build):
+        props = Properties.fromDict(build['properties'])
+        pr_url = props.getProperty("pullrequesturl")
+        match = re.search("^(http|https)://([^/]+)/(.+)$", pr_url)
+
+        if not match:
+            log.error("not valid pull request URL: %s" % (pr_url,))
+            defer.returnValue(None)
+            return
+
+        path = match.group(3)
+        status = "SUCCESS" if build['results']==SUCCESS else "FAILED"
+        props.setProperty('statustext', status, self.name)
+        props.setProperty('url', build['url'], self.name)
+
+        comment_text = yield props.render(self.text)
+        payload = {
+                'text' : comment_text
+                }
+        response = yield self._http.post('/rest/api/1.0/%s/comments' % (path),
```

Comment:
Yes, that looks good.

I agree, if they change the API it's possible lots of things would need to change, but having all the paths at the top makes it much easier to find all the areas of your code that are reading or writing to that API.

I'd probably go further and set:

```
STASH_STATUS_API_URL = '/rest/build-status/1.0/commits/{sha}'
[...]
response = yield self._http.post(STASH_STATUS_API_URL.format(sha=sha), json=payload)
```